### ISSUE TITLE
Abstract Class Support

### ISSUE THREAD

**Issue by CryogenicPlanet** [fasttime/Polytype#11]
Hey

I was trying to use this library to do something a tad bit weird and ran into an error.

`Cannot assign an abstract constructor type to a non-abstract constructor type`

To recreate the bug

```typescript

abstract class AbstractClass {}

class Test extends classes(AbstractClass) {}
// Argument of type 'typeof AbstractClass' is not assignable to parameter of type 'SuperConstructor'.
//  Cannot assign an abstract constructor type to a non-abstract constructor type.ts(2345)
```

Not sure if this is just a typing bug, ie the package supports abstract classes but the types have to be updated or if they package itself doesn't have abstract class support

**Comment by fasttime** [comment]
It's a typing bug. Abstract classes didn't work well in previous versions of Polytype, so they were not allowed as base classes. With TypeScript 4 there shouldn't be any issues.

**Comment by fasttime** [comment]
Fixed in Polytype 0.12.0. Thank you for reporting this issue.
